Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -234,9 +228,18 @@ public Mnemonic CreateWallet(string password, string name, string passphrase = n
                 account.CreateAddresses(this.network, UnusedAddressesBuffer, true);
             }
 
-            // Update the height of the we start syncing from.
-            this.UpdateLastBlockSyncedHeight(wallet, this.chain.Tip);
-
+            // If the chain is downloaded, we set the height of the newly created wallet to it.
+            // However, if the chain is still downloading when the user creates a wallet, 
+            // we wait until it is downloaded in order to set it. Otherwise, the height of the wallet will be the height of the chain at that moment.
+            if (this.chain.IsDownloaded())
+            {
+                this.UpdateLastBlockSyncedHeight(wallet, this.chain.Tip);
+            }
+            else
+            {
+                this.UpdateWhenChainDownloaded(wallet, DateTime.Now);
```

Comment:
My only comment would be to use the `IDateTimeProvider` instead of `DateTime.Now`